flight/Modules/Battery/battery.c

```diff
@@ -128,18 +128,28 @@ static void batteryTask(void * parameters)
 				currentADCPin = -1;
 		}
 
+		bool adc_pin_invalid = false;
+		bool adc_offset_invalid = false;
+
 		// handle voltage
 		if (voltageADCPin >= 0) {
 			flightBatteryData.Voltage = (float)PIOS_ADC_GetChannelVolt(voltageADCPin);
 
 			// A negative result indicates an error (PIOS_ADC_GetChannelVolt returns -1 on error)
 			if(flightBatteryData.Voltage < 0.0f)
-				flightBatteryData.Voltage = 0.0f;
+				adc_pin_invalid = true;
```

Comment:
Honestly, I think the original code is poor in that it is constantly changing a variable's value throughout the function. IMHO, this code would all be cleaner if temporary variables are used. That would make the checks and their intention much more obvious, and thus maintainable.